### PR TITLE
perf(training): trim training metric logging

### PR DESCRIPTION
**Pull request by Borda**
Changes:
- Disable Lightning sanity validation by default while preserving caller overrides.
- Aggregate auxiliary loss metrics and log learning rates at optimizer-update boundaries.
- Cover cadence and metric-schema behavior; document dashboard migration.

Impact:
- Default accumulation of four reduces learning-rate log calls by about 75%.
- Default RFDETRSmall reduces loss-metric state from 17 keys to 9 per microbatch.
- Training dashboards use aggregate auxiliary keys and no longer receive per-layer loss keys.

Verification:
- uv run --no-sync pytest tests/training/test_module_model.py tests/training/test_trainer.py -n 1 -m 'not gpu' --timeout=240 (182 passed, 1 skipped)
- pre-commit run --files CHANGELOG.md docs/learn/train/customization.md src/rfdetr/training/module_model.py src/rfdetr/training/trainer.py tests/training/test_module_model.py tests/training/test_trainer.py (passed)
- uv run --no-sync mkdocs build --strict (passed)
- .reports/codex/implement/2026-08-17T20-47-38-721324Z gates (lint, format, types, tests, review passed)

Residual limits:
- End-to-end throughput and GPU memory effects were not benchmarked; code-derived logging reduction is not a timing result.
- Full-suite terminal output did not retain its final count.
- Full-repository pre-commit still reports pre-existing invalid snippets in docs/getting-started/migration.md.

### CODE REVIEW

File: src/rfdetr/config.py [hunk 1]
```diff
@@ -1234,8 +1234,17 @@ def _coerce_amp_dtype(cls, value: Any) -> Any:
...
     # PTL runtime/perf tuning knobs.
     train_log_sync_dist: bool = False
+    # Component-level train/ metrics honor train_log_on_step for on_step visibility only when
+    # compact_train_metrics=False; when True, components are aggregated and logged on_epoch-only regardless.
     train_log_on_step: bool = False
+    # When True (default), per-decoder/encoder auxiliary loss keys (e.g. loss_ce_0, loss_bbox_enc) are
+    # aggregated into train/<term>_aux and always logged on_epoch-only. When False, every per-layer key is
+    # logged individually, honoring train_log_on_step for those component logs too.
+    compact_train_metrics: bool = True
     compute_train_metrics: bool = False
```
Comment (Borda) [inline src/rfdetr/config.py:1244]:
```suggestion
```
